projects/canopy/src/lib/paginator/paginator.component.ts

```diff
@@ -0,0 +1,272 @@
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  EventEmitter,
+  HostBinding,
+  Inject,
+  InjectionToken,
+  Input,
+  Optional,
+  Output,
+  ViewEncapsulation,
+} from '@angular/core';
+
+import {
+  lgIconChevronLeft,
+  lgIconChevronRight,
+  lgIconGotoFirst,
+  lgIconGotoLast
+} from '../icon/icons.interface';
+
+/**
+ * Event raised when the page index is updated
+ */
+export interface PaginationEvent {
+  /**
+   * The selected page index.
+   */
+  pageIndex: number;
+
+  /**
+   * The selected page size.
+   */
+  pageSize: number;
+
+  /**
+   * The previously selected page index.
+   */
+  previousIndex: number;
+}
+
+/**
+ * New paginator default options.
+ */
+export interface PaginationOptions {
+  /**
+   * The page size.
+   */
+  pageSize?: number;
+
+  /**
+   * The list of page sizes.
+   */
+  pageSizes?: Array<number>;
+
+  /**
+   * Flag to hide the page size selector.
+   */
+  hidePageSize?: boolean;
```

Comment:
For the most part in our existing components, we tend to use `showSomething`, instead of `hideSomething` and then default it to be `true` if desired. I guess in terms of consistency, it'd be nice to keep it that way. I can see below you're using `showFirstLastButtons`.

Reply:
I modeled this on the mat-paginator options https://material.angular.io/components/paginator/api#MatPaginatorDefaultOptions but am happy to go with the consensus.

Reply:
Thanks, Material has been be a very valuable source of inspiration to us too! 🙂 

In my view, it'd be good to be consistent with naming and go with `showSomething` (with a default as `true` if needed) rather than `hideSomething`. But also happy to go with consensus as well.

@thatguynamedandy  @elenagarrone  @aaronBery  @paul-request  - Do you guys have a view on this at all?

Reply:
Apologies, I completely missed the updates on this PR. I agree with @owensgit - for consistency I would use `showSomething` :)